worker/peergrouper/worker_test.go

```diff
@@ -137,194 +137,202 @@ func (s *workerSuite) waitAdvance(c *gc.C, d time.Duration) {
 	s.clock.Advance(d)
 }
 
-func (s *workerSuite) TestSetsAndUpdatesMembers(c *gc.C) {
-	logger.SetLogLevel(loggo.TRACE)
-	loggo.GetLogger("juju.mongo").SetLogLevel(loggo.INFO)
-	loggo.GetLogger("juju.network").SetLogLevel(loggo.INFO)
-	DoTestForIPv4AndIPv6(c, s, func(ipVersion TestIPVersion) {
-		c.Logf("\n\nTestSetsAndUpdatesMembers: %s", ipVersion.version)
-		st := NewFakeState()
-		InitState(c, st, 3, ipVersion)
-		memberWatcher := st.session.members.Watch()
-		mustNext(c, memberWatcher)
-		assertMembers(c, memberWatcher.Value(), mkMembers("0v", ipVersion))
+func (s *workerSuite) TestSetsAndUpdatesMembersIPv4(c *gc.C) {
+	s.dotestSetAndUpdateMembers(c, testIPv4)
```

Comment:
test traceback made it hard to debug, because of the nested helper function.
split it out into 2 tests for each IP format, rather than a Do loop nesting an inner function.

these diffs look a *lot* cleaner if you use "git diff -b" because I changed the indenting, but most of the test didn't actually change.